bin/misc-functions.sh

```diff
@@ -462,38 +462,55 @@ __dyn_package() {
 	# in there in case any tools were built with -pg in CFLAGS.
 	cd "${T}" || die
 
-	local tar_options=""
-	[[ $PORTAGE_VERBOSE = 1 ]] && tar_options+=" -v"
-	has xattr ${FEATURES} && [[ $(tar --help 2> /dev/null) == *--xattrs* ]] && tar_options+=" --xattrs"
+	[[ -z "${BINPKG_FORMAT}" ]] && export BINPKG_FORMAT="xpak"
```

Comment:
Note that environment variables like this are sealed into the environment at the beginning of pkg_setup, in order to ensure continuity of the environment from one ebuild phase to another (refer to "The State of Variables Between Functions" in [PMS](https://dev.gentoo.org/~ulm/pms/head/pms.html#x1-11400011.2)). This is why all of the environment variables used here have a PORTAGE_ prefix and have been whitelisted to break then environment seal via environ_whitelist in lib/portage/package/ebuild/_config/special_env_vars.py.

At the very least, we may want to add BINPKG_FORMAT to the environ_whitelist, to ensure that we are not working with a stale value here.

Reply:
Added to environ_whitelist and added some values checks.